lib/config-new.js

```diff
@@ -0,0 +1,524 @@
+   */
+  serverJobsAbandonedTimeoutSec: z.number().default(30),
+  /**
+   * Controls whether or not the course request form will attempt to automatically
+   * create a course if the course request meets certain criteria.
+   */
+  courseRequestAutoApprovalEnabled: z.boolean().default(false),
+  attachedFilesDialogEnabled: z.boolean().default(true),
+  devMode: z.boolean().default((process.env.NODE_ENV ?? 'development') === 'development'),
+  /** The client ID of your app in AAD; required. */
+  azureClientID: z.string().default('<your_client_id>'),
+  /** The reply URL registered in AAD for your app. */
+  azureRedirectUrl: z.string().default('<your_redirect_url>'),
+  /** Required if the redirect URL uses the HTTP protocol. */
+  azureAllowHttpForRedirectUrl: z.boolean().default(false),
+  /** Required. If the app key contains `\`, replace it with `\\`. */
+  azureClientSecret: z.string().default('<your_client_secret>'),
+  /**
+   * Required to encrypt cookies. Multiple key/iv pairs can be provided for key
+   * rotation. The first key/iv pair will be used to encrypt cookies, but all
+   * key/iv pairs will be used to attempt to decrypt cookies.
+   *
+   * The key must have length 32, and the iv must have length 12.
+   */
+  azureCookieEncryptionKeys: z
+    .array(
+      z.object({
+        key: z.string().length(32),
+        iv: z.string().length(12),
+      })
+    )
+    .default([]),
+  azureLoggingLevel: z.enum(['error', 'warn', 'info']).default('warn'),
+  /**
+   * If you want to get access_token for a specific resource, you can provide the
+   * resource here; otherwise, set the value to null.
+   * Note that in order to get access_token, the responseType must be 'code', 'code id_token' or 'id_token code'.
+   */
+  azureResourceURL: z.string().nullable().default('https://graph.windows.net'),
+  /**
+   * The URL to which the user will be redirected to destroy the session.
+   */
+  azureDestroySessionUrl: z
+    .string()
+    .nullable()
+    .default(
+      'https://login.microsoftonline.com/common/oauth2/logout?post_logout_redirect_uri=http://localhost:3000'
+    ),
+});
+
+/** @typedef {z.infer<typeof ConfigSchema>} Config */
+
+const loader = new ConfigLoader(ConfigSchema);
+
+module.exports.config = loader.config;
+
+module.exports.loadConfig = async function (path) {
+  await loader.loadAndValidate([
+    makeFileConfigSource(path),
+    makeImdsConfigSource(),
+    makeSecretsManagerConfigSource('ConfSecret'),
+  ]);
+};
+
+module.exports.ConfigSchema = ConfigSchema;
+
+module.exports.setLocalsFromConfig = (locals) => {
+  locals.homeUrl = module.exports.config.homeUrl;
+  locals.urlPrefix = module.exports.config.urlPrefix;
+  locals.plainUrlPrefix = module.exports.config.urlPrefix;
+  locals.navbarType = 'plain';
+  locals.devMode = module.exports.config.devMode;
+  locals.is_administrator = false;
+};
+
+module.exports.getDBConfigValue = async function (key, defaultVal) {
```

Comment:
I *think* in favor of that, yeah.

`display_timezone` will be *slightly* annoying since we use the `config_select` sproc to get that value inside of some other queries; we'll have to switch to manually piping that around. But even then, honestly, I think we could get away with displaying these sort of global dates (those not attached to something in an institution/course/course instance) in UTC. We're already in a situation where I (PT) have to do math to go from CT to either PT or UTC; I think it'd be fine for *all* admins to have to do UTC math since there's no guarantee on what timezones we'll be in. Do you buy that?

`workspaceAutoscalingEnabled` would be a great use case for "negative" feature flags - I think there's value in having a kill switch for this that doesn't require a redeploy to take effect. I suppose we could always disable the auto scaling process on the ASG itself if we really need to stop scaling...

The only other thing we reference from this table is `workspaceDesiredHostCount`. I don't think we've ever had to use that to pin the ASG to a certain size?

Reply:
I'm totally fine with switching global admin to be UTC. The server-level `display_timezone` was a leftover from when we only had one institution, I think.

For the ASG, I think we can stop autoscaling manually by setting max instances = min instances in the ASG console, so it presumably doesn't really matter what the code is doing? I don't think we've ever actually used this kill switch in practice. Similarly, I think `workspaceDesiredHostCount` can be overridden by ASG instance settings, should we ever need it.